packages/main/cypress/specs/DatePicker.cy.tsx

```diff
@@ -0,0 +1,1622 @@
+			.ui5DatePickerGetFirstDisplayedDate()
+			.invoke("attr", "data-sap-timestamp")
+			.should("include", timestamp_31_Dec_0000);
+	});
+
+	it("daypicker next extreme values max", () => {
+		const timestamp_28_Nov_9999 = "253399363200";
+
+		cy.mount(<DatePicker value="Nov 30, 9999"></DatePicker>);
+
+		cy.get("[ui5-date-picker]")
+			.as("datePicker")
+			.ui5DatePickerValueHelpIconPress();
+
+		cy.get<DatePicker>("@datePicker")
+			.ui5DatePickerGetNextButton()
+			.realClick();
+
+		cy.get<DatePicker>("@datePicker")
+			.ui5DatePickerGetFirstDisplayedDate()
+			.invoke("attr", "data-sap-timestamp")
+			.should("include", timestamp_28_Nov_9999);
+	});
+
+	it("monthpicker next extreme values max", () => {
+		cy.mount(<DatePicker value="Dec 31, 9998"></DatePicker>);
+
+		cy.get("[ui5-date-picker]")
+			.as("datePicker")
+			.ui5DatePickerValueHelpIconPress();
+
+		cy.get<DatePicker>("@datePicker")
+			.ui5DatePickerGetMonthButton()
+			.realClick();
+
+		cy.get<DatePicker>("@datePicker")
+			.ui5DatePickerGetNextButton()
+			.realClick();
+
+		cy.get<DatePicker>("@datePicker")
+			.shadow()
+			.find("ui5-calendar")
+			.as("calendar")
+			.shadow()
+			.find("ui5-monthpicker")
+			.should("be.visible");
+
+		cy.get<DatePicker>("@datePicker")
+			.ui5DatePickerGetYearButton()
+			.invoke("text")
+			.should("include", "9999");
+	});
+
+	it("monthpicker prev extreme values min", () => {
+		cy.mount(<DatePicker value="Jan 1, 0002"></DatePicker>);
+
+		cy.get("[ui5-date-picker]")
+			.as("datePicker")
+			.ui5DatePickerValueHelpIconPress();
+
+		cy.get<DatePicker>("@datePicker")
+			.ui5DatePickerGetMonthButton()
+			.realClick();
+
+		cy.get<DatePicker>("@datePicker")
+			.shadow()
+			.find("ui5-calendar")
+			.as("calendar")
+			.shadow()
+			.find("ui5-monthpicker")
+			.should("be.visible");
+
+		cy.get<DatePicker>("@datePicker")
+			.ui5DatePickerGetPreviousButton()
+			.realClick();
+
+		cy.get<DatePicker>("@datePicker")
+			.ui5DatePickerGetYearButton()
+			.invoke("text")
+			.should("include", "0001");
+	});
+
+	it("yearpicker extreme values max", () => {
+		cy.mount(<DatePicker value="Dec 31, 9995"></DatePicker>);
+
+		cy.get("[ui5-date-picker]")
+			.as("datePicker")
+			.ui5DatePickerValueHelpIconPress();
+
+		cy.get<DatePicker>("@datePicker")
+			.ui5DatePickerGetYearButton()
+			.realClick();
+
+		cy.get<DatePicker>("@datePicker")
+			.ui5DatePickerGetFirstDisplayedYear()
+			.invoke("text")
+			.should("include", "9980");
+	});
+
+	it("yearpicker extreme values min", () => {
+		cy.mount(<DatePicker value="Jan 1, 0003"></DatePicker>);
+
+		cy.get("[ui5-date-picker]")
+			.as("datePicker")
+			.ui5DatePickerValueHelpIconPress();
+
+		cy.get<DatePicker>("@datePicker")
+			.ui5DatePickerGetYearButton()
+			.realClick();
+
+		cy.get<DatePicker>("@datePicker")
+			.ui5DatePickerGetFirstDisplayedYear()
+			.invoke("text")
+			.should("include", "0001");
+	});
+
+	it("yearpicker prev page extreme values min", () => {
```

Comment:
I think in the tests with extremes (min/max dates hitted 0001/9999 years) we should also assert that the next/prev button in calendar headers are disabled, as we recently saw similar bug in UI5.